### ISSUE TITLE
Skins: Load via sampler button

### ISSUE THREAD

**Issue by daschuer** [mixxxdj/mixxx#13918]
### Feature Description

In the mappings we have the feature to load samplers with the sampler button:
https://github.com/mixxxdj/mixxx/blob/2f04ffcdf7b1db7a633897fbe8bba89838ca495a/res/controllers/midi-components-0.0.js#L379-L386

I find my selef clicking on the empty sample button in skins as well. However it does not work. Id does also not work for xml mappings for the samper button. 

These mappins are affected: 
https://github.com/search?q=repo%3Amixxxdj%2Fmixxx+cue_gotoandplay+language%3AXML+path%3A%2F%5Eres%5C%2Fcontrollers%5C%2F%2F&type=code

A solution that makes this feature consistenly availabel for all would be nice.

**Comment by ronso0** [comment]
Are you suggesting to implement the components logic in https://github.com/mixxxdj/mixxx/blob/894e96f9b27d182bcc8ad779db8599056a93006e/src/engine/controls/cuecontrol.cpp#L1347
with a switch `if (!m_pLoadedTrack && m_isSampler)` and `m_isSampler` being set in the ctor with `PlayerManager::isSamplerGroup(group)`?

**Comment by Swiftb0y** [comment]
Lets add a new CO for this...

**Comment by daschuer** [comment]
My original idea was to put it here: 

https://github.com/mixxxdj/mixxx/blob/894e96f9b27d182bcc8ad779db8599056a93006e/src/engine/enginebuffer.cpp#L759

Than it will work for any action that tries to play an empty deck/sampler. 

Both other suggestions work as well. In case of a new CO, we need to touch all skins and mappings. 

Can this be ever undesired? 
Is there a situation we don't want this feature?

**Comment by ronso0** [comment]
> Can this be ever undesired?
Is there a situation we don't want this feature?

Hmm, it might happen that I press Play on a **deck** assuming I still have a certain track loaded, but it's already been ejected.
Then press Play won't do anything (except load a track), which makes me wonder what's wrong. Then I discover a probably unexpected track loaded in the deck, which requires more steps to get (back) to the state I want.
(well, we have 'un-replace' for such cases)

IMO the scenario for samplers is usually different. On controllers the sampler's play Button is usually lit when a track is loaded, so that state is more clear. And in skins we see the track title.

I don't have a strong opinion here.
Might be a nice helper in skins for both samplers and main decks, but there we could as well implement it with a special Play button overlay, which has the benefit of being able to show a dedicated icon.
But I wouldn't change the general behaviour of Play (all payers), just sampler and preview decks.

**Comment by daschuer** [comment]
Yes that makes sense. So we need to exclude decks and consider GUI feedback ... A special Widget for the sampler button maybe?

**Comment by ronso0** [comment]
Sure, we can already put a button with `[SamplerN],LoadSelectedtrack` on top, hidden when a track is loaded.
Or were you thinking about a new WWidget?

**Comment by daschuer** [comment]
In that case I was thinking about a new widget. But I don't mind which solution we pick finally. 

For the controllers a new CO is probably reasonable. Sometimes linke "sampler_default" this can be also used in skins. This would allow to adjust specific behaviors for different use cases from preferences.
